docs/validator.md

```diff
@@ -0,0 +1,205 @@
+dereferenced.
+
+# Components
+
+## gpii.schema.validator.ajv
+
+The common grade, designed for use on the client and server-side.
+
+### Component Options
+
+The following component configuration options are supported:
+
+| Option             | Type     | Description |
+| ------------------ | -------- | ----------- |
+| `validatorOptions` | `Object` | Options to be passed to our AJV validator instance (check [the AJV options documentation](https://github.com/epoberezkin/ajv#options) for the syntax and available options). |
+
+
+### Invokers
+
+#### `{validator}.validate(key, content)`
+
+* `key {String}`:  The schema key we are validating against.
+* `content {Object}`: The JSON content we are validating.
+* Returns: An `{Object}` describing the errors, or `undefined` if there are no errors.
+
+The raw output provided by AJV is run through `sanitizeValidationErrors` (see below) to provide more user-friendly
+feedback where possible.
+
+### Static Functions
+
+
+#### `gpii.schema.validator.ajv.sanitizeValidationErrors(schemaKey, errors)`
+
+* `schemaKey {String}`: The schema key we are validating against.
+* `errors`: A map of existing error results to be sanitized (see below for examples).
+
+AJV gives us output like:
+
+```
+ [
+     {
+         "keyword": "minLength",
+         "schemaPath": "/password/minLength",
+         "dataPath": ".password",
+         "message": "should NOT be shorter than 8 characters"
+     },
+     {
+         "keyword": "pattern",
+         "schemaPath": "/password/allOf/1/pattern",
+         "dataPath": ".password",
+         "message": "should match pattern \"[A-Z]+\""
+     },
+     {
+         "keyword": "required",
+         "schemaPath": "/deep/required/0",
+         "dataPath": ".deep.required",
+         "message": "is a required property"
+     }
+ ]
+```
+
+This function uses [`gpii.schema.errors.evolveError`](evolveErrors.md) to look for `errors` definitions that match the `schemaPath` output
+from the validator. If it finds them, it replaces `message` with alternate wording, so that we end up with output like:
+
+```
+ [
+     {
+         "keyword": "minLength",
+         "schemaPath": "/password/minLength",
+         "dataPath": ".password",
+         "message": "Passwords must be at least 8 characters long."
+     },
+     {
+         "keyword": "pattern",
+         "schemaPath": "/password/allOf/1/pattern",
+         "dataPath": ".password",
+         "message": "Passwords must contain at least one uppercase character."
+     },
+     {
+         "keyword": "required",
+         "schemaPath": "/deep/required/0",
+         "dataPath": ".deep.required",
+         "message": "You must enter a value for the 'deeply required' field."
+     }
+ ]
+```
+See [the documentation for `evolveError`](evolveErrors.md) for more details.
+
+Note that `dataPath` values are ["Javascript property access notation"](https://github.com/epoberezkin/ajv#validation-errors).
```

Comment:
Thanks for your researches in discovering this. It appears that this is actually part of a public standard, https://tools.ietf.org/html/rfc6901 , which you should reference. 

A helpful courtesy (need not be part of this pull request) would be to supply a utility that can convert this boneheaded format into something which can be supplied to the ChangeApplier API. Most helpfully, something which will parse both schemaPath and dataPath into an array of path segments which will then make escaping issues moot. A future version of this library could accompany these by schemaSegs and dataSegs in these records holding these parsed values.